### PR TITLE
Secure ShellExecute/DLLImport

### PR DESCRIPTION
**Pull request by smitpatel**
Resolves https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1712534/

###### Microsoft Reviewers: [Open in CodeFlow](https://portal.fabricbot.ms/api/codeflow?pullrequest=https://github.com/dotnet/project-system/pull/8816)

### CODE REVIEW

**Review by MiYanni — DISMISSED** [review]

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/WindowsFileExplorer.cs [hunk 1]
```diff
@@ -25,7 +25,10 @@ public void OpenContainingFolder(string path)
             if (_fileSystem.PathExists(path))
             {
                 // Tell Explorer to open the parent folder of the item, selecting the item
-                ShellExecute(string.Empty, "explorer.exe", parameters: $"/select,\"{path}\"");
+                ShellExecute(
+                    string.Empty,
+                    Environment.GetFolderPath(Environment.SpecialFolder.System) + Path.DirectorySeparatorChar + "explorer.exe",
```
[the remark this answers is not in the document]
Comment (drewnoakes) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/WindowsFileExplorer.cs:30]: `Path.Combine` for .NET Framework.

**Comment by drewnoakes** [comment]
Why not also in `Microsoft.VisualStudio.ProjectSystem.Managed.VS`?

I think we should set `DefaultDllImportSearchPathsAttribute` on every assembly via `Directory.Build.props` to prevent this issue popping up in future if we add another assembly.

We'd need both a `GlobalAssemblyInfo.cs` and `GlobalAssemblyInfo.vb` file, each containing the attribute, then something like this in `Directory.Build.props` to link them to projects based on language:

```xml
<ItemGroup>
  <Compile Include="GlobalAssemblyInfo.cs" Condition="'$(Language)' == 'C#'" />
  <Compile Include="GlobalAssemblyInfo.vb" Condition="'$(Language)' == 'VB'" />
</ItemGroup>
```

File: src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/WindowsFileExplorer.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (MiYanni) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/WindowsFileExplorer.cs:30]: Ah, right. There was someone telling me to use `Path.Join` before and I couldn't remember why I told them we use `Path.Combine` in our code. That's why.

[the remark this answers is not in the document]
Comment (MiYanni) [reply src/Microsoft.VisualStudio.ProjectSystem.Managed/IO/WindowsFileExplorer.cs:30]: Found it. This conversation here; https://devdiv.visualstudio.com/DefaultCollection/DevDiv/_git/dotnet-project-system-vscode/pullRequest/440969#1670872739

**Comment by smitpatel** [comment]
I went where DLLImport was present but I like your idea that putting at global level will save us from making error in future. I will update the PR.

**Comment by smitpatel** [comment]
Updated.

**Review by MiYanni — APPROVED** [review]
LGTM :shipit:

**Pull request merged**